### PR TITLE
ScalametaParser: handle leading infix correctly

### LINKED ISSUE #2880: ParseException `; expected but . found`
input :

```scala
    Flow.fromGraph(GraphDSL.create() {
      implicit b =>
        import GraphDSL.Implicits._

        val removeItems       = b.add(RemoveLineItems(settings.lineItems.removal).graph)
        val filtering         = b.add(Filtering(settings.filtering).flow)
        val projection        = b.add(Projection(settings.projection).graph)

        val input_<       = b.add(Broadcast[FileContent](2))

        // format: off
        input_< ~> filtering          ~>       removeItems.in0
        // format: on

        FlowShape(input_<.in, projection.out)
    })
```

dialect : `scala213source3`
scalameta : `v4.5.13`
scala : `2.13.8`

### PR DESCRIPTION
**Pull request by kitbellew**
https://docs.scala-lang.org/scala3/reference/changed-features/operators.html#syntax-change

Fixes #2880. Supersedes #2888.

### CODE REVIEW

File: tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/InfixSuite.scala [hunk 1]
```diff
@@ -397,21 +419,24 @@ class InfixSuite extends BaseDottySuite {
          |""".stripMargin,
       Some(
         """|{
-           |  println("hello")
-           |  ???
-           |  ??? match {
+           |  println("hello") ??? ??? match {
```
Comment (kitbellew) [inline tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/InfixSuite.scala:422]: this is the example mentioned just above unary operators (https://docs.scala-lang.org/scala3/reference/changed-features/operators.html#unary-operators) but the explanation there is unclear: it says `???` is a symbolic _identifier_, hence the second `???` can -- and seemingly does -- start an expression; is there another reason why it couldn't?

**Review by tgodzik — COMMENTED** [review]
Thanks for working on this! I had hard time getting back to the issue. I have some small questions and also should we not keep some of the tests that we had previously instead of changing them?

File: scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala [hunk 2]
```diff
@@ -2031,6 +2029,32 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
         }
       }
 
+      def tryGetNextRhsIfLeading: Option[Either[Term, Term]] =
+        tryAhead(token match {
+          case opToken: Ident if opToken.isIdentSymbolicInfixOperator =>
+            val op = atCurPosNext(Term.Name(opToken.value))
```
[the remark this answers is not in the document]
Comment (kitbellew) [reply scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala:2035]: this turns a token into a tree

File: tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/InfixSuite.scala [hunk 3]
```diff
@@ -160,7 +183,10 @@ class InfixSuite extends BaseDottySuite {
          |""".stripMargin,
       Some(
         """|Flow {
-           |  def foo = (b.add() input_< ~> filtering ~> removeItems1 ~>).removeItems2
```
[the remark this answers is not in the document]
Comment (kitbellew) [reply tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/InfixSuite.scala:163]: could you explain, please? this expression is not the input to the test but the expected syntax value after parsing. as the parsing has been fixed, so will the expected syntax change.

also, the input is valid, shouldn't fail but should be interpreted differently.

File: tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/InfixSuite.scala [hunk 4]
```diff
@@ -232,11 +255,7 @@ class InfixSuite extends BaseDottySuite {
          |  || xs.isEmpty
          |""".stripMargin,
       Some(
-        """|def condition = {
-           |  x > 0
-           |  ||
-           |  xs.exists(_ > 0) || xs.isEmpty
-           |}
+        """|def condition = x > 0 || xs.exists(_ > 0) || xs.isEmpty
```
[the remark this answers is not in the document]
Comment (kitbellew) [reply tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/InfixSuite.scala:258]: same idea: this is the expected parsec syntax.

**Review by kitbellew — COMMENTED** [review]
i added expected syntax, in addition to expected tree, as a more user-friendly way to see how it was parsed.

File: tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/InfixSuite.scala (hunk 4 above)
[the remark this answers is not in the document]
Comment (tgodzik) [reply tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/InfixSuite.scala:258]: Ach, right! Thanks for explaining!

File: tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/InfixSuite.scala (hunk 3 above)
[the remark this answers is not in the document]
Comment (tgodzik) [reply tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/InfixSuite.scala:163]: Right! I got a bit lost in the changes it seems.

[the remark this answers is not in the document]
Comment (kitbellew) [reply tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/InfixSuite.scala:163]: sorry about the confusion! and thanks for reviewing.

**Pull request merged**